### ISSUE TITLE
Better documentation for the HTML report

### ISSUE THREAD

**Issue by fofoni** [python/mypy#13495]
**Documentation**

[This](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-html-report) is the only thing I can find in the docs about this HTML report, and it's far from complete. I could find [these](https://stackoverflow.com/questions/50891468/legenda-or-documentation-for-the-mypy-html-report) [three](https://stackoverflow.com/questions/67112890/understanding-mypy-coverage-report-coloring-any-types-omitted-generics) [questions](https://stackoverflow.com/questions/61307000/deciphering-python-mypy-reports) on Stack Overflow, but none of them have satisfying answers (or answers at all).

I understand that mypy is reporting the presence of expressions of type `Any` in each line. (Is it? Did I get that right?) But what exactly do the colors mean? Why do lines with `@overload` always get colored red in my reports? Why do lines with `isinstance` always get colored yellow?

**Comment by wyattscarpenter** [comment]
Duplicate of #912, I think. But anyone who fixes that should keep in mind to document it in a way that answers your questions!

**Comment by wyattscarpenter** [comment]
I was able to replicate your problems on mypy master (`mypy 2.0.0+dev.3cda6ec255a530e7da8bec44d197399aec73e96d (compiled: no)`), such as the `@overload` thing. I think this is a bug, especially considering that [the typeshed definition of overload](https://github.com/python/typeshed/blob/4567c8c5284f0608202aea612f3fcc1a190b97a9/stdlib/typing.pyi#L410) does not contain `Any`, but it is hard for me to say definitively (due to the lack of documentation).

With the following code:

```py
from typing import overload

def foo(x):
    return x

def foo1(x: int) -> int:
    return x

def foo2(x) -> int:
    return x

def foo3(x: int):
    return x

def foo4(x: int):
    return x

@overload
def foo5(x: int) -> int: ...
@overload
def foo5(x: str) -> str: ...
def foo5(x):
    return x

@overload
def foo6(x: int) -> int: ...
@overload
def foo6(x: str) -> str: ...
def foo6(x: int|str) -> int|str:
    return x
```

And the following command:
```sh
uv run mypy --html-report tmp_dir tmp.py
```

I get the following result:

<img width="269" height="568" alt="Image" src="https://github.com/user-attachments/assets/c6ef2614-8d0b-47f0-88ab-c8e52d60b6e6" />

The `@overload` lines are red but on hover say "No Anys on this line!". Surprisingly, this is also true of `def foo(x):` and `def foo5(x):`

My `isinstance` lines are always colored green, not yellow. In fact, the line `isinstance(1,list)` is green, but on hover it says `Any Types on this line: Omitted Generics (x3)`. I think this kind of makes sense as behavior given the rules about `isinstance` but I think in that case the hovertext should maybe be different.

In the following code, the `print("OK")` line is colored red despite on hover saying `No Anys on this line!`. The `if isinstance(v,list):` line is green despite saying `Any Types on this line: Omitted Generics (x3)`

```py
v = 1
if isinstance(v,list):
    print("OK")
```
 
Possibly related: https://github.com/python/mypy/issues/14297
